apps/server/src/modules/files-storage/uc/files-storage.uc.ts

```diff
@@ -126,14 +132,9 @@ export class FilesStorageUC {
 	}
 
 	public async uploadFromUrl(userId: EntityId, params: FileRecordParams & FileUrlParams) {
-		await this.checkPermission(
-			userId,
-			params.parentType,
-			params.parentId,
-			FileStorageAuthorizationContext.create(params.storageLocation)
-		);
+		await this.checkPermission(userId, params.parentType, params.parentId, FileStorageAuthorizationContext.create);
 
-		await this.checkStorageLocation(params.storageLocation, params.storageLocationId);
+		await this.checkStorageLocation(userId, params.storageLocation, params.storageLocationId);
```

Comment:
await Promise.all([
this.checkPermission(userId, params.parentType, params.parentId, FileStorageAuthorizationContext.create),
this.checkStorageLocation(userId, params.storageLocation, params.storageLocationId)
]);